### PR TITLE
CDAP-15813: Vertica export plugin validations.

### PR DESCRIPTION
**Pull request by AntonGensitskiy**
JIRA: https://issues.cask.co/browse/CDAP-15813
WIKI: https://wiki.cask.co/display/CE/Guidelines+on+Plugin+Validation

In scope of this PR:

- integrated with new validation API

### CODE REVIEW

**Comment by AntonGensitskiy** [comment]
@albertshau @CuriousVini could you, please, take a look.

File: pom.xml [hunk 1]
```diff
@@ -144,8 +144,11 @@
           <extensions>true</extensions>
           <configuration>
             <instructions>
-              <_exportcontents>io.cdap.plugin.*;org.apache.commons.lang;org.apache.commons.logging.*;
-                org.codehaus.jackson.*; </_exportcontents>
+              <_exportcontents>
+                io.cdap.plugin.db.batch.action.vertica.export.*;
+                org.apache.commons.lang;org.apache.commons.logging.*;
```
Comment (albertshau) [inline pom.xml]: this was an existing problem, but actions are completely self contained and don't need to export these other packages. Please remove them.

File: src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+  public void setDelimiter(@Nullable String delimiter) {
+    this.delimiter = delimiter;
+  }
+
+  public String getPath() {
+    return path;
+  }
+
+  public void setPath(String path) {
```
Comment (albertshau) [inline src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java]: there is a builder, so these setters should be removed.

File: src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+    this.path = path;
+  }
+
+  public void validate(FailureCollector failureCollector) {
+    if (Strings.isNullOrEmpty(path)) {
+      failureCollector.addFailure("File Path is required.","File path must be specified.")
```
Comment (albertshau) [inline src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java]: Just have 'File Path must be specified' as the error message and use a null corrective action.

File: src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+        .withConfigProperty(PATH);
+    }
+    if (Strings.isNullOrEmpty(connectionString)) {
+      failureCollector.addFailure("Connection String is required.",
```
Comment (albertshau) [inline src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java]: similar comment here

File: src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+                                  "Connection String must be specified.")
+        .withConfigProperty(CONNECTION_STRING);
+    } else if (!connectionString.startsWith(CONNECTION_STRING_PREFIX)) {
+      failureCollector.addFailure(
+        "Connection String has incorrect format.",
+        "Connection String must comply next format - jdbc:vertica://<VerticaHost>:<portNumber>/<databaseName>")
```
Comment (albertshau) [inline src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java]: Ensure the connection string is of format jdbc:vertica:...

File: pom.xml (hunk 1 above)
Comment (AntonGensitskiy) [reply pom.xml]: Done.

File: src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java (hunk 2 above)
Comment (AntonGensitskiy) [reply src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java]: Done.

File: src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java (hunk 3 above)
Comment (AntonGensitskiy) [reply src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java]: Done.

File: src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java (hunk 4 above)
Comment (AntonGensitskiy) [reply src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java]: Done.

File: src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java (hunk 5 above)
Comment (AntonGensitskiy) [reply src/main/java/io/cdap/plugin/db/batch/action/vertica/export/VerticaExportConfig.java]: Done.

**Comment by AntonGensitskiy** [comment]
@albertshau I addressed all your comments, could you, please, take a look.

File: pom.xml [hunk 6]
```diff
@@ -144,11 +144,6 @@
           <extensions>true</extensions>
           <configuration>
             <instructions>
-              <_exportcontents>
-                io.cdap.plugin.db.batch.action.vertica.export.*;
```
Comment (albertshau) [inline pom.xml]: this line still needs to be exported (it's where the plugin lives). The 2 lines below are the ones that are not needed

**Review by albertshau — COMMENTED** [review]
one comment, otherwise lgtm. please fix and squash.

File: pom.xml (hunk 6 above)
Comment (AntonGensitskiy) [reply pom.xml]: Done.

**Comment by AntonGensitskiy** [comment]
@albertshau pom.xml was updated

**Comment by albertshau** [comment]
please squash the commits and i'll merge

**Comment by AntonGensitskiy** [comment]
Commits are squashed

**Pull request merged**